Dispatcher/UI/Signup - Login/AuthViewController.swift

```diff
@@ -0,0 +1,371 @@
+import UIKit
+
+enum eyeIconStatus {
+    case conseal
+    case reveal
+}
+
+enum SignupLoginPageCurrentView {
+    case signup
+    case login
+}
+
+class AuthViewController: UIViewController, LoadingViewDelegate {
+
+    @IBOutlet weak var loadingView: LoadingView!
+    
+    @IBOutlet weak var logoImageView: UIImageView!
+    
+    @IBOutlet weak var titleLabel: UILabel!
+    
+    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var wrongEmailLabel: UILabel!
+
+    @IBOutlet weak var passwordTextField: UITextField!
+    @IBOutlet weak var passwordEyeIcon: UIImageView!
+    @IBOutlet weak var weakPasswordLabel: UILabel!
+    
+    @IBOutlet weak var reenterPasswordTextField: UITextField!
+    @IBOutlet weak var reenterPasswordEyeIcon: UIImageView!
+    @IBOutlet weak var mismatchPasswordLabel: UILabel!
+    
+    @IBOutlet weak var separatorLine: UIView!
+    
+    @IBOutlet weak var topButton: MainActionButtonView!
+    @IBOutlet weak var bottomButton: MainActionButtonView!
+    
+    
+    
+    let viewModel = AuthViewModel()
+    var currentPageType: SignupLoginPageCurrentView = .signup
+    var passwordEyeStatus: eyeIconStatus = .conseal
+    var reenterPasswordEyeStatus: eyeIconStatus = .conseal
+    
+    var separatorConstraintLogin: NSLayoutConstraint? = nil
+    var separatorConstraintSignup: NSLayoutConstraint? = nil
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        recognizeUser()
+        
+        initializeUIElements()
+        defineGestureRecognizers()
+        defineConstraints()
+    }
+    
+    func recognizeUser() {
+        
+        startLoadingScreen()
+        
+        switch viewModel.checkIfLoggedIn() {
+        case .freshUser:
+            currentPageType = .signup
+        case .loggedOut:
+            currentPageType = .login
+        case .loggedIn:
+            DispatchQueue.main.async {
+                self.currentPageType = .login
+                self.performSegue(withIdentifier: Constants.Segues.AUTH_SCREEN_TO_APP, sender: self)
+            }
+        }
+        
+        stopLoadingScreen()
+    }
+    
+    func initializeUIElements() {
+        
+        loadingView.initView(delegate: self)
+        showHideElements()
+        setTextFields()
+        setActionButtons()
+
+        
+        let screenHeight = UIScreen.main.bounds.height
+        logoImageView.heightAnchor.constraint(equalToConstant: CGFloat(screenHeight / 2 )).isActive = true
+
+        
+        if currentPageType == .login {
+            setLoginPageLook()
+        } else {
+            setSignupPageLook()
+        }
+    }
+    
+    func showHideElements() {
+        loadingView.isHidden = true
+        wrongEmailLabel.isHidden = true
+        weakPasswordLabel.isHidden = true
+        mismatchPasswordLabel.isHidden = true
+    }
+    
+    func setTextFields() {
+        emailTextField.delegate = self
```

Comment:
Again, looks like we're repeating ourself here. Strong sign to do it as custom view.